e2e/e2e_test.go

```diff
@@ -368,3 +372,100 @@ func TestPVCResize(t *testing.T) {
 	}
 	steps.Run(t)
 }
+func TestRollingRestart(t *testing.T) {
+
+	// Restart cluster nodes in a rolling cadence, one after the other
+
+	if parallel {
+		t.Parallel()
+	}
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	actor.Log = testLog
+
+	sb := testenv.NewDiffingSandbox(t, env)
+	sb.StartManager(t, controller.InitClusterReconcilerWithLogger(testLog))
+
+	builder := testutil.NewBuilder("crdb").WithNodeCount(3).WithTLS().
+		WithImage("cockroachdb/cockroach:v20.2.5").
+		WithPVDataStore("1Gi", "standard" /* default storage class in KIND */)
+
+	steps := Steps{
+		{
+			name: "creates a 3-node secure cluster",
+			test: func(t *testing.T) {
+				require.NoError(t, sb.Create(builder.Cr()))
+
+				RequireClusterToBeReadyEventuallyTimeout(t, sb, builder, 500*time.Second)
+			},
+		},
+		{
+			name: "restarts the cluster using the rolling functionality",
+			test: func(t *testing.T) {
+				current := builder.Cr()
+				require.NoError(t, sb.Get(current))
+
+				pods, err := fetchPodsInStatefulSet(sb, labels.Common(current).Selector())
+				require.NoError(t, err)
+				require.NotNil(t, pods)
+
+				podCreationTimes := make(map[string]metav1.Time)
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "-vcheck-") {
+						continue // Skip the validate version pod
+					}
+					podCreationTimes[pod.Name] = pod.CreationTimestamp
+				}
+
+				current.Annotations["crdb.io/restarttype"] = "rolling"
+				require.NoError(t, sb.Update(current))
+
+				// Need to poll for getting rid of the restart annotation
+				err = wait.Poll(10*time.Second, 500*time.Second, func() (done bool, err error) {
```

Comment:
I think this is likely to fail once we add the enhanced restart logic because of a timeout issue